micro-s3/src/main/java/com/aol/micro/server/s3/DirectoryCleaner.java

```diff
@@ -0,0 +1,35 @@
+package com.aol.micro.server.s3;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DirectoryCleaner {
+
+	private final String temporaryDirectory;
+	private final boolean cleanupOnStart;
+
+	@Autowired
+	public DirectoryCleaner(@Value("${s3.temp.dir:#{null}") String temporaryDirectory,
+			@Value("${s3.temp.cleanupOnStart:false") boolean cleanupOnStart) {
+		this.temporaryDirectory = temporaryDirectory;
+		this.cleanupOnStart = cleanupOnStart;
```

Comment:
this variable should only be true if the tmp.dir setting is not null and exists.